packages/gatsby-cli/src/create-cli.js

```diff
@@ -105,6 +105,25 @@ function buildLocalCommands(cli, isLocalSite) {
     handler: getCommandHandler(`develop`),
   })
 
+  cli.command({
```

Comment:
Are you assuming that you'd have `gatsby develop` running and then run this in another terminal tab? If so, this wouldn't work as Gatsby keeps the node data in memory — we need to add the ability to pass commands to the running gatsby develop process e.g. press "r" and a refresh would be triggered.

Reply:
This does work (in my local tests, at least), as it makes a post request to the port the server is running on, hitting an endpoint that calls `sourceNodes()`.

If I understand correctly, you envision that the process running `gatsby develop` is also listening to `stdin` and updating based on input there, (eg: pressing "r" in the terminal window). I'm not familiar with any programs that do that, but I can totally give that a shot.

Even without the `refresh` command, would you be open to the dev server accepting a webhook to reload the data? That is my use case and what some others commenting on the related issues are looking for.

Reply:
ah, pressing `r` as in reload (kind of like how Jest CLI works?) could be nice for local development, but I agree with @PepperTeasdale that the main use case would be to use the webhook … since it's just a POST, building a trigger that fits the requirements on top of it would be trivial.